contracts/governance/extensions/GovernorProposalGuardian.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+
+/**
+ * @dev Extension of {Governor} which adds a proposal guardian that can cancel proposals at any stage of their lifecycle.
+ *
+ * Note: if the proposal guardian is not configured, then proposers take this role for their proposals.
+ */
+abstract contract GovernorProposalGuardian is Governor {
+    address private _proposalGuardian;
+
+    event ProposalGuardianSet(address oldProposalGuardian, address newProposalGuardian);
+
+    /**
+     * @dev Override {IGovernor-cancel} that implements the extended cancellation logic.
+     * * proposal guardian can cancel any proposal at any point in the lifecycle.
+     * * if no proposal guardian is set, proposer can cancel their proposals at any point in the lifecycle.
+     * * if the proposal guardian is set, proposer keep their ability to cancel during the pending stage (default behavior).
```

Comment:
Although the default behavior in the Governor is that a proposal can be canceled by the proposer, I would argue the default behavior is to call `super`. The difference is that `super` may call another override up in the inheritance chain.

```suggestion
     * * if the proposal guardian is set, the {proposalProposer} keeps their default rights defined in {IGovernor-cancel} (calling `super`).
```